### PR TITLE
Fixes #175: quantizer failing to build a palette with transparent col…

### LINKED ISSUE #175: Quantizer can sometimes remove transparent color from palette
The current color sampling alghoritm does not consider a completely transparent color "special", so it can be removed from the final palette and transparent pixels can become "almost" transparent (low alpha, random color)

### PR DESCRIPTION
**Pull request by mbarto**
…or sometimes

### CODE REVIEW

File: jt-colorindexer/src/main/java/it/geosolutions/jaiext/colorindexer/Quantizer.java [hunk 1]
```diff
@@ -88,6 +95,12 @@ public ColorIndexer buildColorIndexer(RenderedImage image) {
 
         // setup the first box, that median cut will split in parts
         List<Box> boxes = new ArrayList<Box>();
+        if (histogram.hasTransparentPixels()) {
+            BufferedImage transparentImage = new BufferedImage(1, 1, BufferedImage.TYPE_INT_ARGB);
+            transparentImage.getRaster().setPixel(0, 0, new int[] {0, 0, 0, 0});
+            PackedHistogram transparentHisto = new PackedHistogram(transparentImage, subsx, subsy);
```
Comment (dromagnoli) [inline jt-colorindexer/src/main/java/it/geosolutions/jaiext/colorindexer/Quantizer.java]: I'm wondering if we could have this "utility" PackedHistogram being statically initialized

File: jt-colorindexer/src/main/java/it/geosolutions/jaiext/colorindexer/Quantizer.java [hunk 2]
```diff
@@ -37,6 +44,8 @@
 public class Quantizer {
 
     static final Logger LOGGER = Logger.getLogger("Quantizer");
+    
+    static PackedHistogram transparentHisto;
```
Comment (aaime) [inline jt-colorindexer/src/main/java/it/geosolutions/jaiext/colorindexer/Quantizer.java]: static final PackedHistogram TRANSPARENT_HISTO;

File: jt-colorindexer/src/main/java/it/geosolutions/jaiext/colorindexer/Quantizer.java [hunk 3]
```diff
@@ -88,6 +103,9 @@ public ColorIndexer buildColorIndexer(RenderedImage image) {
 
         // setup the first box, that median cut will split in parts
         List<Box> boxes = new ArrayList<Box>();
+        if (histogram.hasTransparentPixels()) {
+            boxes.add(new Box(0, 1, 1, transparentHisto, null));
```
Comment (aaime) [inline jt-colorindexer/src/main/java/it/geosolutions/jaiext/colorindexer/Quantizer.java]: Should it update the "colors" count too? I have a hunch the change as is might fail on solid color tiles (e..g, a fully transparent tile)

File: jt-colorindexer/src/main/java/it/geosolutions/jaiext/colorindexer/Quantizer.java [hunk 4]
```diff
@@ -19,12 +19,19 @@
...
+import java.awt.image.Raster;
 import java.awt.image.RenderedImage;
+import java.awt.image.SampleModel;
+import java.awt.image.WritableRaster;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
+import java.util.Vector;
```
Comment (aaime) [inline jt-colorindexer/src/main/java/it/geosolutions/jaiext/colorindexer/Quantizer.java]: Unused import here (java.awt.Rectangle too, not sure about the others)

**Comment by aaime** [comment]
+1

**Pull request merged**